scripts/webidl/convert.js

```diff
@@ -1,56 +1,23 @@
 "use strict";
-const fs = require("fs");
-const path = require("path");
-const Q = require("q");
-const readdirRecursive = require("fs-readdir-recursive");
-const webidl2js = require("webidl2js");
-
-const outputDir = path.resolve(__dirname, "../../lib/jsdom/living/generated/");
-Q.longStackSupport = true;
-
-function readConcatenatedSource(files) {
-  return Q.all(files.map(f => {
-    return Q.nfcall(fs.readFile, f, { encoding: "utf8" });
-  })).then(sources => {
-    let src = "";
-    for (let i = 0; i < sources.length; ++i) {
-      src += sources[i];
-    }
-    return src;
-  });
-}
 
-function generateClasses(src, implDir) {
-  webidl2js.generate(src, outputDir, implDir, { suppressErrors: true, implSuffix: "-impl" });
-}
+const path = require("path");
 
-function doConversion(inputPath) {
-  let isDir;
+const Webidl2js = require("webidl2js");
 
-  return Q.nfcall(fs.stat, inputPath)
-  .then(inputStat => {
-    isDir = inputStat.isDirectory();
-    if (isDir) {
-      return readdirRecursive(inputPath, onlyIDL).map(relativePath => path.resolve(inputPath, relativePath));
-    }
+const transformer = new Webidl2js();
+transformer.options.suppressErrors = true;
```

Comment:
This seems like a bad API; the options should be passed in to the constructor, I think.